prowler/providers/azure/services/app/app_service.py

```diff
@@ -112,3 +182,16 @@ class WebApp:
     https_only: bool = False
     monitor_diagnostic_settings: list[DiagnosticSetting] = None
     kind: str = "app"
+
+
+@dataclass
+class FunctionApp:
+    name: str
+    location: str
+    kind: str
+    function_keys: Dict[str, str]
+    enviroment_variables: Dict[str, str]
+    identity: ManagedServiceIdentity
+    public_access: bool
+    vnet_subnet_id: str
```

Comment:
This attribute cannot store a None value if `virtual_network_subnet_id` does not exist.